src/main/java/com/dongsoop/dongsoop/chat/validator/ChatValidator.java

```diff
@@ -113,27 +173,27 @@ private Set<String> extractMessageIds(List<ChatMessage> messages) {
                 .collect(Collectors.toSet());
     }
 
-    private List<ChatMessage> filterMessages(List<ChatMessage> messages, Predicate<ChatMessage> filter) {
-        return messages.stream()
-                .filter(filter)
+    private List<ChatMessage> filterNewMessages(List<ChatMessage> clientMessages, Set<String> existingIds) {
+        Predicate<ChatMessage> isNewMessage = msg -> !existingIds.contains(msg.getMessageId());
+
+        return clientMessages.stream()
+                .filter(isNewMessage)
                 .toList();
     }
 
-    public void validateManagerPermission(ChatRoom room, Long userId) {
-        Long managerId = room.getManagerId();
-        if (managerId != null && !managerId.equals(userId)) {
-            throw new UnauthorizedManagerActionException();
-        }
+    private void validateUserExistsInRoom(ChatRoom room, Long userToKick) {
+        Optional.of(!room.getParticipants().contains(userToKick))
+                .filter(Boolean::booleanValue)
+                .ifPresent(ignored -> {
+                    throw new UserNotInRoomException();
+                });
     }
```

Comment:
불필요한 래핑을 줄이는 것이 좋아보입니다.

```suggestion
    public void validateManagerPermission(ChatRoom room, Long userId) {
        Long managerId = room.getManagerId();
        if (managerId != null && !managerId.equals(userId)) {
            throw new UnauthorizedManagerActionException();
        }
    }
```